lib/hash.js

```diff
@@ -0,0 +1,71 @@
+'use strict'
+
+const crypto = require('node:crypto')
+const fs = require('node:fs/promises')
+const path = require('node:path')
+const os = require('node:os')
+const { Glob } = require('glob')
+const fastq = require('fastq')
+
+async function generateFileHash (filePath) {
+  try {
+    const fileBuffer = await fs.readFile(filePath)
+    return `${crypto.createHash('md5').update(fileBuffer).digest('hex').slice(0, 16)}-`
+  } catch {
+    return ''
+  }
+}
+
+/**
+ * Generates hashes for files matching the glob pattern within specified directories.
+ * Can be used both at build time and runtime.
+ *
+ * @param {string|string[]} rootPaths - A single root directory or an array of root directories to process.
+ * @param {boolean} [includeDotFiles] - Whether to include dot files.
+ * @param {string|string[]} [ignore] - An array of glob patterns to ignore.
+ * @param {boolean} [writeToFile] - Whether to write the hash map to a file (for build) or return it (for runtime).
+ * @param {string} [outputPath] - The output file path, if writing to a file.
+ * @returns {Promise<void|Map<string, string>>} - Returns nothing if writing to a file, or the hash map if not.
+ */
+async function generateHashes (rootPaths, includeDotFiles = false, ignore = ['node_modules/**'], writeToFile = false, outputPath) {
```

Comment:
nit: this function takes a lot of arguments that are kinda hard to tell what they are just by looking at the callsite. What do you think about making it take an options object, kind of like `Glob` itself does? Would make it a bit easier to pass through all the options as well